README.md

```diff
@@ -95,6 +99,64 @@ vmm-reference                           \
     --kernel path=/path/to/kernel/image
 ```
 
+## Testing
+
+The reference VMM is, first and foremost, a vehicle for end-to-end testing of
+`rust-vmm` crates. Each crate must contain individual functional and
+performance tests that exercise as wide a range of use cases as possible; the
+reference VMM is not meant to reiterate on that, but to validate all the pieces
+put together.
+The Rust unit tests are testing modules in isolation and private interfaces,
+while the two Rust integration tests (one for each supported kernel image
+format, i.e. ELF and bzImage) exercise the only public function of the `Vmm`
+object, `run()`.
+The Python integration tests make use of the VMM in varied configurations that
+aren’t overly complex and illustrate realistic use cases (e.g. one test runs a
+VMM with a virtio block device, one test will run a VMM with PCI, etc.).
+
+To be able to successfully run all the tests in this repo, there are a couple of
+resources that need to be found in `/tmp`. To generate the resources at the
+expected locations, you can run the following command:
+
+```bash
+sudo ./resources/make_test_resources.sh
+```
+
+`vmm-reference` is a
+(workspace)[https://doc.rust-lang.org/book/ch14-03-cargo-workspaces.html], so to
+run all the Rust tests, the following command should be used:
+
+```bash
+cargo test --workspace
+```
+
+There is no single command yet for running all the Python integration tests in
+one shot. To run the tests from a single file, you can use:
+
+```bash
+pytest-3 <path_to_the_file>
+```
+For example:
+```bash
```

Comment:
nit: surround code snippets with blank lines (here and below)

Reply:
Yep it's a not-exactly-official guideline from [here](https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md#md031---fenced-code-blocks-should-be-surrounded-by-blank-lines), and also one of the things the [`mdl` markdown linter](https://github.com/markdownlint/markdownlint) complains about